projects/packages/connection-ui/src/class-admin.php

```diff
@@ -12,6 +12,13 @@
  */
 class Admin {
 
+	/**
+	 * Make sure we only initialize the class once, as multiple plugins may call `init()` here.
+	 *
+	 * @var $is_initialized
+	 */
+	public static $is_initialized = false;
```

Comment:
Sorry I'm late to the singleton party!
So, looking at this code it doesn't seem to be that different from a Singleton. Although the property is public, which makes sense for possible testing/mocking, but doesn't really make sense in the way it's used. 

So I would like to propose something that we don't regularly do, but I think we could look at doing more often. We are initializing this in the WordPress environment, and WordPress has hooks, which is a great tool.

What if we did something like this in the constructor:

```
if ( ! did_action( 'jetpack_on_connection_ui_init' ) ) {
    // add_action here
    do_action( 'jetpack_on_connection_ui_init' );
}
```

This has a nice side effect of adding a possibility for other consumers to know when exactly the UI was initialized and hooks added to all the right places, in case they need to be modified/amended for whatever reason.

Reply:
Thanks Igor, that's a great suggestion and an option Kraft mentioned to me as well, I like it!